seloger-scraper/seloger.py

```diff
@@ -0,0 +1,99 @@
+"""
+This is an example web scraper for seloger.com.
+
+To run this scraper set env variable $SCRAPFLY_KEY with your scrapfly API key:
+$ export $SCRAPFLY_KEY="your key from https://scrapfly.io/dashboard"
+"""
+import os
+import json
+from scrapfly import ScrapeConfig, ScrapflyClient, ScrapeApiResponse
+from typing import Dict, List
+from pathlib import Path
+from loguru import logger as log
+
+SCRAPFLY = ScrapflyClient(key=os.environ["SCRAPFLY_KEY"])
+
+BASE_CONFIG = {
+    "asp": True,
+    "country": "fr",
+}
+
+output = Path(__file__).parent / "results"
+output.mkdir(exist_ok=True)
+
+
+def parse_search(result: ScrapeApiResponse):
+    """parse property listing data from seloger search pages"""
+    # select the script tag from the HTML
+    script_jsons = result.selector.xpath("//script[contains(., 'window[\"initialData\"]')]/text()").re(
+        r'JSON.parse\("(.+?)"\)'
+    )
+    # decode the JSON data
+    datasets = [json.loads(script_json.encode("utf-8").decode("unicode_escape")) for script_json in script_jsons]
+    cards = []
+    # validate the cards data to avoid advertisement cards
+    for card in datasets[0]["cards"]["list"]:
+        if len(card) > 12:
```

Comment:
so the cards with <= 12 datapoint keys are ads? Maybe there's a better way to identify this? Seems a bit flimsy.